diffpy/nmf_mapping/nmf_mapping/nmf_mapping_code.py

```diff
@@ -237,7 +239,7 @@ def component_plot(df_components, xrd=False, x_units=None, show=True):
     """
     try:
```

Comment:
this is also a bit ugly.  It think that we just need to fix the requirements so that this works and remove the try/except.